### PR TITLE
[CPM-43] Add Edit Capability

### PR DESCRIPTION
**Pull request by daltonlim**
## Changes

- User now able to edit years/name of a plan
- Can view the plans for a given student
- Can edit plan from student page 
- Ability to delete a plan yet to be implemented

## Checklist

- [x] Have tests been written?
- [ ] Is the UI consistent?

## Requests
- [x] Can somebody fix the styling/CSS?

Addresses [CPM-43]

Connects #y

### CODE REVIEW

**Comment by FocalChord** [comment]
@pull-dog down

File: client/src/pages/Plan/Year.js [hunk 1]
```diff
@@ -52,6 +53,7 @@ const CourseTile = ({ courseName, courses }) => {
                 {getCourseRegulations()}
             </ReactTooltip>
             <Icon name="info-outline" color={c.white} data-tip data-for={courseName} />
+            <Icon name="close" color={c.white} onClick={() => updateData(data.map((data) => data.name != courseName))} />
```
Comment (Richardng12) [inline client/src/pages/Plan/Year.js:56]: Tests failing cuz its not data.name !== courseName

**Review by Richardng12 — COMMENTED** [review]
image
image

I get these error when i create a new student, as they dont have a plan currently assigned to them

**Comment by FocalChord** [comment]
I think Nasser mentioned he want's a notification that his information is being saved everytime a change happens. You could use a Toast to do that I reckon (whenever an API call is made)

**Comment by daltonlim** [comment]
> I think Nasser mentioned he want's a notification that his information is being saved everytime a change happens. You could use a Toast to do that I reckon (whenever an API call is made)

@FocalChord can we make a new ticket for that?

**Comment by daltonlim** [comment]
@jira CPM-43

**Comment by daltonlim** [comment]
@jira[bot] CPM-43

File: client/src/pages/Plan/Year.js [hunk 2]
```diff
@@ -115,7 +115,11 @@ const Year = ({ year, data, startYear, updateData, courses, setStartYear }) => {
     return (
         <Flex direction="column" mt={4}>
             <Flex width="100%" justify="center" align="center" marginTop="20px">
-                <InlineEdit title="Year" value={startYear + year} onChange={(e) => setStartYear(e - year)} />
+                <InlineEdit
+                    title="Year"
+                    value={startYear + year}
+                    onChange={(e) => new RegExp("^[0-9]+$").test(e) && setStartYear(e - year)}
```
Comment (FocalChord) [inline client/src/pages/Plan/Year.js:121]: Beautiful line of code

Comment (daltonlim) [reply client/src/pages/Plan/Year.js:121]: gotta prevent the user from entering letters :p

**Review by Richardng12 — COMMENTED** [review]
LGTM functionality wise, just need styling on the buttons

**Comment by daltonlim** [comment]
@pull-dog up

**Comment by FocalChord** [comment]
@pull-dog up

**Review by FocalChord — APPROVED** [review]
lgtm

**Review by Richardng12 — APPROVED** [review]
lgtm

**Pull request merged**